src/BcPrebidVast.js

```diff
@@ -230,33 +243,127 @@ function loadPrebidScript(options, fromHeader) {
+			// failed to load prebid.js.
+			_logger.error(_prefix, 'Failed to load prebid.js. Error event: ', e);
+			if (options.pageNotificationCallback) {
+				options.pageNotificationCallback('message', 'Failed to load prebid.js');
+			}
+			$$PREBID_GLOBAL$$.bc_pbjs_error = true;
+			dispatchPrebidDoneEvent();
+		};
+		pbjsScr.id = 'bc-pb-script';
+		pbjsScr.async = true;
+		pbjsScr.type = 'text/javascript';
+		pbjsScr.src = options.prebidPath ? options.prebidPath : DEFAULT_PREBID_JS_URL;
+		var node = document.getElementsByTagName('head')[0];
+		node.appendChild(pbjsScr);
+	}
+	else {
+		// plugin has been loaded in the document body or has been embedded in a player
+		var frame = document.createElement('iframe');
+		frame.id = 'pbjs_conainer_' + Date.now();
+		frame.src = 'about:blank';
+		frame.marginWidth = '0';
+		frame.marginHeight = '0';
+		frame.frameBorder = '0';
+		frame.width = '0%';
+		frame.height = '0%';
+		frame.style.position = 'absolute';
+		frame.style.left = '0px';
+		frame.style.top = '0px';
+		frame.style.margin = '0px';
+		frame.style.padding = '0px';
+		frame.style.border = 'none';
+		frame.style.width = '0%';
+		frame.style.height = '0%';
+		document.body.appendChild(frame);
+
+		var template = defaultTemplate;
+		template = template.replace(new RegExp('{{iframePrebid_JS}}', 'g'), options.prebidPath ? options.prebidPath : DEFAULT_PREBID_JS_URL);
+		template = template.replace(new RegExp('{{origin}}', 'g'), getOrigin());
+
+		var timeout = setTimeout(function() {
+			// failed to load prebid.js in iframe.
+			_logger.error(_prefix, 'Failed to load prebid.js in iframe (timeout).');
+			if (options.pageNotificationCallback) {
+				options.pageNotificationCallback('message', 'Failed to load prebid.js in iframe (timeout)');
+			}
+			$$PREBID_GLOBAL$$.bc_pbjs_error = true;
+			dispatchPrebidDoneEvent();
+			timeout = null;
+		}, 3000);
+
+		var iframeDoc = frame.contentWindow && frame.contentWindow.document;
+		if (iframeDoc) {
+			try {
+				iframeDoc.open();
+				iframeDoc.write(template);
+				iframeDoc.close();
+				var onLoad = function(msgEvent) {
+					if (timeout) {
+						clearTimeout(timeout);
+					}
+					else {
+						// prebid.js loadding timeout already happened. do nothing
+						return;
+					}
+					// check only our messages 'ready' and 'error' from ifarme
+					if (msgEvent.data === 'ready') {
+						$$PREBID_GLOBAL$$.bc_pbjs = frame.contentWindow.pbjs;
+						// after prebid.js is successfully loaded try to invoke prebid.
+						doInternalPrebid();
+					}
+					else if (msgEvent.data === 'error') {
+						// failed to load prebid.js.
+						_logger.error(_prefix, 'Failed to load prebid.js in iframe.');
+						if (options.pageNotificationCallback) {
+							options.pageNotificationCallback('message', 'Failed to load prebid.js inframe');
+						}
+						$$PREBID_GLOBAL$$.bc_pbjs_error = true;
+						dispatchPrebidDoneEvent();
+					}
+				};
+
+				window.addEventListener('message', onLoad);
```

Comment:
Is there ever a cleanup routine for this event listener? I fear adding the 'message' listener but never clearing it will lead to a memory leak resulting from an orphaned anonymous listener reference.

I suggest you should store the `onLoad` function in a permanent property, and do a `window.removeEventListener` whenever you destroy the plugin instance.

Reply:
Have sense. Will fix it.

Reply:
Sounds good.